From 6d7a32e3f3a95d900d44c0619621d359039d6f97 Mon Sep 17 00:00:00 2001
From: "Radu Pirea (NXP OSS)" <radu-nicolae.pirea@oss.nxp.com>
Date: Thu, 10 Feb 2022 15:05:48 +0200
Subject: [PATCH 28/43] arch: s32: call debug_uart_init from arch_cpu_init

Call debug_uart_init from arch_cpu_init.

Issue: ALB-8504

Upstream-Status: Pending 

Signed-off-by: Radu Pirea (NXP OSS) <radu-nicolae.pirea@oss.nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/cpu.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/cpu/armv8/s32/cpu.c b/arch/arm/cpu/armv8/s32/cpu.c
index 3171cf5456..3d923e88e8 100644
--- a/arch/arm/cpu/armv8/s32/cpu.c
+++ b/arch/arm/cpu/armv8/s32/cpu.c
@@ -14,6 +14,9 @@
 #include <asm/arch/soc.h>
 #include <asm-generic/sections.h>
 #include <linux/sizes.h>
+#if defined(CONFIG_DEBUG_UART)
+#include <debug_uart.h>
+#endif
 
 #define S32GEN1_DRAM_STD_ADDR	0x80000000ULL
 #define S32GEN1_DRAM_EXT_ADDR	0x800000000ULL
@@ -211,6 +214,9 @@ int arch_cpu_init(void)
 	__asm_invalidate_tlb_all();
 	early_mmu_setup();
 
+#if defined(CONFIG_DEBUG_UART)
+	debug_uart_init();
+#endif
 	return ret;
 }
 
-- 
2.17.1

